### ISSUE TITLE
Visualization Problem

### ISSUE THREAD

**Issue by howardyclo** [yuanming-hu/fc4#40]
Hi Yuanming,

Thanks for the released code. I've been having fun playing around with your code.
Since I just recently studied about color constancy, I have a (probably stupid )problem about the visualization of corrected image.

The corrected image is rendered as follows:
```
img_corrected = tf.pow(
      images[:, :, :, ::-1] / 65535 / illums_pooled[:, None, None, :] * exposure_boost *
      tf.reduce_mean(illums_pooled, axis=(1), keep_dims=True)[:, None, None, :],
      1 / VIS_GAMMA)
```

Here what I do not understand about is that why do we need this ` * tf.reduce_mean(illums_pooled...)`? I surveyed several papers and I couldn't find the white-balancing formula about this.

**Comment by yuanming-hu** [comment]
Hi,

You can consider `tf.reduce_mean(illums_pooled, axis=(1), keep_dims=True)[:, None, None, :]/llums_pooled[:, None, None, :]` together. This normalizes the illumination estimation and makes sure the output image is not too dark/bright.

Note that color constancy cares about only color but not about brightness.
